airgun/views/contentviewfilter.py

```diff
@@ -0,0 +1,343 @@
+            )
+            select_all = Checkbox(
+                locator=".//table//th[@class='row-select']/input")
+            table = SatTable(locator=".//table")
+
+            def search(self, query=None, filters=None):
+                """Custom search which supports all errata filters.
+
+                :param str optional query: search query to type into search
+                    box. Optional as sometimes filtering is enough to find
+                    desired errata
+                :param dict optional filters: dictionary containing widget
+                    names and values to set (like with regular `fill()`)
+                """
+                if isinstance(filters, dict):
+                    for key, value in filters.items():
+                        getattr(self, key).fill(value)
+                if query:
+                    self.searchbox.search(query)
+                return self.table.read()
+
+            def add(self, errata_id=None, filters=None):
+                """Add specific errata to filter or all available if id not
+                provided.
+
+                :param str optional errata_id: ID of errata to add. If not
+                    provided - all available errata in table will be selected
+                    (especially useful together with filtering)
+                :param dict optional filters: dictionary containing widget
+                    names and values to set (like with regular `fill()`)
+                """
+                self.search(errata_id, filters)
+                if errata_id:
+                    self.table.row((
+                        'Errata ID', errata_id))[0].widget.fill(True)
+                else:
+                    self.select_all.fill(True)
+                self.add_button.click()
+
+            def fill(self, errata_id=None, filters=None):
+                self.add(errata_id, filters)
+
+            def read(self):
+                return self.table.read()
+
+        @View.nested
+        class erratum_date_range(SatSecondaryTab):
+            TAB_NAME = 'Erratum Date Range'
+
+            security = Checkbox(locator=".//input[@ng-model='types.security']")
+            enhancement = Checkbox(
+                locator=".//input[@ng-model='types.enhancement']")
+            bugfix = Checkbox(
+                locator=".//input[@ng-model='types.bugfix']")
+            date_type = RadioGroup(
+                ".//div[label[contains(@class, 'radio-inline')]]")
+            start_date = DatePicker(
+                locator=".//input[@ng-model='rule.start_date']")
+            end_date = DatePicker(
+                locator=".//input[@ng-model='rule.end_date']")
+
+            save = Text('//button[contains(@ng-click, "handleSave()")]')
+            cancel = Text('//button[contains(@ng-click, "handleCancel()")]')
+
+            def after_fill(self, was_change):
+                self.save.click()
+
+        def add(self, errata_id=None, filters=None):
```

Comment:
you already specified add and fill for `AddTab`, so why do you implement it once more with small deviation?

Reply:
Following `AddRemoveResourcesView` pattern, and since we can't reuse original `add()`method due to different signature, the only way is to override this method as well.